app/controllers/application_controller.rb

```diff
@@ -53,56 +41,31 @@ def require_admin
     handle_error(message)
   end
 
-  def html_request?
-    request.format.symbol == :html
-  end
-
-  def api_request?
-    [:json].include? request.format.symbol
-  end
-
-  def api_key
-    request.headers['HTTP_API_KEY']
-  end
-
-  def github_id_from_api_key(api_key)
-    return nil if api_key.nil?
+  private
 
-    client = Octokit::Client.new(access_token: api_key)
-    client.user.id
-  rescue Octokit::Unauthorized => e
-    raise UnauthorizedError::GitHubAuthenticationError, "Error authenticating via GitHub: #{e.message}"
-  end
+  def authenticator
+    if @authenticator.nil?
+      @authenticator = if html_request?
+                         WebAuthenticator.new(self)
+                       elsif api_request?
+                         ApiAuthenticator.new(self)
+                       else
+                         fail UnauthorizedError, "Unable to authenticate"
+                       end
+    end
 
-  def redirect_if_not_logged_in!
-    should_redirect = !current_user
-    redirect_to '/login' if should_redirect
-    should_redirect
+    @authenticator
   end
 
-  # rubocop:disable Style/AccessorMethodName
-  def set_current_user_to_api_user!(raise_errors: false)
-    user = User.where(github_id: github_id).first
 
-    fail UnauthorizedError::UserNotFound if user.nil? && raise_errors
-    @current_user = user
+  def html_request?
```

Comment:
These methods are just used in the `authenticator` factory above so I might just inline them there